addons/point_of_sale/models/product.py

```diff
@@ -277,14 +277,24 @@ class ProductTemplateAttributeLine(models.Model):
     def _load_pos_data_fields(self, config_id):
         return ['display_name', 'attribute_id', 'product_template_value_ids']
 
+    @api.model
+    def _load_pos_data_domain(self, data):
+        loaded_product_tmpl_ids = list({p['product_tmpl_id'] for p in data['product.product']['data']})
+        return AND([[('product_tmpl_id', 'in', loaded_product_tmpl_ids)]])
```

Comment:
do you need the `AND` here ?

Reply:
@vlst-odoo no it's not needed. I modified the code.